decentraland/unity-explorer
# feat: third party wearables

## What does this PR change?

Adds the support for third party wearables v1 & v2. Shape definition: https://www.notion.so/decentraland/LinkedWearables-Migration-ffaf57a6dcc84390b2b8c8232d87fb28

- Adds the param when fetching wearables in the backpack to retrieve those which are third party in the grid.
- Supports the new urn schema
- Applies a refactor on the backpack so wearables are not retrieved directly through ecs intentions. Instead there is an abstraction layer to hide its complexity
- Includes a `--dclenv` program arg so either `zone`/`org` can be changed in the build, and we can properly test `sepolia` network
- Implements a refactor of the program args processing improving the classes responsibilities
- Adds a service to fetch the third party providers information: `{{catalystUrl}}/lambdas/third-party-integrations`
- Adds a new field in the backpack item info panel to display the third party collection name
- Fixes an endless loading for wearables `type:facialFeature` that dont have a mask texture. Specially when it was required to fallback to the default wearable `urn:decentraland:off-chain:base-avatars:eyes_00`
- Face snapshot texture size changed to 256x256 when the profile is published as part of the catalysts requirements: https://github.com/decentraland/content-validator/blob/d82589e6271591bb515a77a553755ef5bf1955ab/src/validations/profile.ts#L40

<img width="1110" alt="Screenshot 2024-08-09 at 14 34 46" src="https://github.com/user-attachments/assets/b328bdae-ff0b-4dc7-a31c-ac5ae6e19342">

## How to test the changes?



1. Mint a third party wearable either v1 and v2
2. Those in sepolia network should be tested executing the client with this params: `--dclenv zone --realm https://peer.decentraland.zone`. Also make sure to change your metamask to sepolia when signing the login. If you are already logged-in, then re-login.
3. Open the backpack
4. All wearables should be displayed in the grid, including TPW
5. Equip TPW
6. Close the backpack
7. Restart the client
8. The profile should keep the equipped TPW
9. Use the backpack filters, they should be working as expected
10. Change to non-TPW wearables. The profile should be saved as expected

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
